controllers/helmchart_controller.go

```diff
@@ -373,37 +375,71 @@ func (r *HelmChartReconciler) reconcileFromHelmRepository(ctx context.Context,
 		return sourcev1.HelmChartNotReady(chart, sourcev1.ChartPullFailedReason, err.Error()), err
 	}
 
+	var valuesFiles []string
+	// Prepend a list of valuesFiles with the deprecated ValuesFile.
+	if chart.Spec.ValuesFile != "" {
+		valuesFiles = append(valuesFiles, chart.Spec.ValuesFile)
+	}
+	valuesFiles = append(valuesFiles, chart.Spec.ValuesFiles...)
+
 	// Either repackage the chart with the declared default values file,
 	// or write the chart directly to storage.
 	var (
 		readyReason  = sourcev1.ChartPullSucceededReason
 		readyMessage = fmt.Sprintf("Fetched revision: %s", newArtifact.Revision)
 	)
 	switch {
-	case chart.Spec.ValuesFile != "" && chart.Spec.ValuesFile != chartutil.ValuesfileName:
+	case len(valuesFiles) > 0:
 		var (
 			tmpDir  string
 			pkgPath string
 		)
+		valuesMap := make(map[string]interface{})
+
 		// Load the chart
 		helmChart, err := loader.LoadArchive(res)
 		if err != nil {
 			err = fmt.Errorf("load chart error: %w", err)
 			return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
 		}
 
-		// Find override file and retrieve contents
-		var valuesData []byte
-		cfn := filepath.Clean(chart.Spec.ValuesFile)
-		for _, f := range helmChart.Files {
-			if f.Name == cfn {
-				valuesData = f.Data
-				break
+		for _, v := range valuesFiles {
+			if v == "values.yaml" {
+				valuesMap = transform.MergeMaps(valuesMap, helmChart.Values)
+				continue
+			}
```

Comment:
Default values are not available from the filesystem, only via `helmChart.Values`

Reply:
Please note this @relu